pyccel/parser/syntactic.py

```diff
@@ -538,25 +520,13 @@ def _visit_Constant(self, stmt):
             return LiteralComplex(stmt.value.real, stmt.value.imag)
 
         elif isinstance(stmt.value, str):
-            return self._visit_Str(stmt)
+            if isinstance(self._context[-2], ast.Expr):
+                return CommentBlock(stmt.s)
+            return LiteralString(stmt.s)
```

Comment:
I think `stmt.s` is part of the `Str` interface but not the `Constant` interface, so it raises the deprecation warning.